tst/unit/test_pararrays.cpp

```diff
@@ -0,0 +1,430 @@
+  SECTION("md range") {
+    std::cout << "md range:" << std::endl;
+    profile_wrapper_3d(parthenon::loop_pattern_mdrange_tag);
+  }
+  SECTION("tpttr") {
+    std::cout << "tpttr range:" << std::endl;
+    profile_wrapper_3d(parthenon::loop_pattern_tpttr_tag);
+  }
+  SECTION("tpttrtvr") {
+    std::cout << "tpttrvr range:" << std::endl;
+    profile_wrapper_3d(parthenon::loop_pattern_tpttrtvr_tag);
+  }
+  */
+#ifndef KOKKOS_ENABLE_CUDA
+  /*
+  SECTION("tptvr") {
+    std::cout << "tptvr range:" << std::endl;
+    profile_wrapper_3d(parthenon::loop_pattern_tptvr_tag);
+  }
+  */
+  SECTION("simdfor") {
+    std::cout << "simd range:" << std::endl;
+    profile_wrapper_3d(parthenon::loop_pattern_simdfor_tag);
+  }
+#endif // !KOKKOS_ENABLE_CUDA
+}
+
+TEST_CASE("Check registry pressure","[ParArrayND]") {
+  auto exec_space = DevSpace();
+  Kokkos::Timer timer;
+
+  // https://en.cppreference.com/w/cpp/numeric/random/uniform_real_distribution
+  std::random_device
+    rd; // Will be used to obtain a seed for the random number engine
+  std::mt19937 gen(rd()); // Standard mersenne_twister_engine seeded with rd()
+  std::uniform_real_distribution<Real> dis(-1.0, 1.0);
+
+  // view of views. See:
+  // https://github.com/kokkos/kokkos/wiki/View#6232-whats-the-problem-with-a-view-of-views
+  using arrays_t = Kokkos::View<ParArrayND<Real>*,UVMSpace>;
+  using views_t = Kokkos::View<ParArray3D<Real>*,UVMSpace>;
+  using device_view_t = parthenon::device_view_t<Real>;
+  arrays_t arrays (Kokkos::view_alloc(std::string("arrays"),
+                                      Kokkos::WithoutInitializing),
+                   NARRAYS);
+  views_t views (Kokkos::view_alloc(std::string("views"),
+                                    Kokkos::WithoutInitializing),
+                 NARRAYS);
+  for (int n = 0; n < NARRAYS; n++) {
+    std::string label = std::string("array ") + std::to_string(n);
+    new (&arrays[n])
+      ParArrayND<Real>(device_view_t(Kokkos::view_alloc(label,
+                                                        Kokkos::WithoutInitializing),
+                                           1,1,1,N,N,N));
+    label = std::string("view ") + std::to_string(n);
+    new (&views[n]) ParArray3D<Real>(Kokkos::view_alloc(label,
+                                                        Kokkos::WithoutInitializing),
+                                     N,N,N);
+    auto a_h = arrays(n).GetHostMirror();
+    auto v_h = Kokkos::create_mirror_view(views(n));
+    for (int k = 0; k < N; k++) {
+      for (int j = 0; j < N; j++) {
+        for (int i = 0; i < N; i++) {
+          a_h(k,j,i) = dis(gen);
+          v_h(k,j,i) = dis(gen);
+        }
+      }
+    }
+    Kokkos::deep_copy(views(n),v_h);
+    arrays(n).DeepCopy(a_h);
+  }
+  // perform a compute intensive, non-stencil, task and see
+  // performance
+  Kokkos::fence();
+  timer.reset();
+  parthenon::par_for(parthenon::loop_pattern_flatrange_tag,
```

Comment:
I don't like that we have to specify the loop pattern, in my mind this should be automatically chosen at compile time depending on the architecture. This isn't necessary to change in this commit though.

Reply:
Yeah, the loop pattern is set to a default in `parthenon/src/CMakeLists.txt`. If we want to set the default for tests too, we can move it to `parthenon/CMakeLists.txt`. @pgrete what are your thoughts on that?

Reply:
Personally, I'd vote for a default setting that is defined through the parameter file and realized as a member variable in each MeshBlock (yelling at people if they don't use our recommended default setting for a specific architecture).
Thus, only keep the most flexible `par_for`s in `kokkos_abstraction.hpp` and specialize at the `MeshBlock` level.
But this may be a longer discussion, see existing #68, that doesn't need to (more like should not) be addressed in this PR.